cmd/lncli/cmd_chain.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/lightningnetwork/lnd/lnrpc/chainrpc"
+	"github.com/urfave/cli"
+)
+
+func getChainClient(ctx *cli.Context) (chainrpc.ChainKitClient, func()) {
+	conn := getClientConn(ctx, false)
+
+	cleanUp := func() {
+		conn.Close()
+	}
+
+	return chainrpc.NewChainKitClient(conn), cleanUp
+}
+
+var getBlockCommand = cli.Command{
+	Name:     "getblock",
+	Category: "On-chain",
+	Usage:    "Get block by block hash.",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "hash",
+			Usage: "the target block hash",
+		},
+	},
+	Description: "Returns a block given the corresponding block hash.",
+	Action:      actionDecorator(getBlock),
+}
+
+func getBlock(ctx *cli.Context) error {
+	ctxc := getContext()
+	client, cleanUp := getChainClient(ctx)
+	defer cleanUp()
+
+	if !ctx.IsSet("hash") {
+		return fmt.Errorf("hash argument missing")
+	}
+	blockHash, err := hex.DecodeString(ctx.String("hash"))
+	if err != nil {
+		return err
+	}
+
+	req := &chainrpc.GetBlockRequest{BlockHash: blockHash}
+	resp, err := client.GetBlock(ctxc, req)
+	if err != nil {
+		return err
+	}
+
+	// Convert raw block bytes into wire.MsgBlock.
+	msgBlock := &wire.MsgBlock{}
+	blockReader := bytes.NewReader(resp.RawBlock)
+	err = msgBlock.Deserialize(blockReader)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("%s", spew.Sdump(msgBlock))
```

Comment:
I don't think we actually want the whole block to be printed in the CLI. Maybe just the header? And then add an optional `--verbose` method that prints a JSON output of the full block (or maybe the raw block as hex)?

The `spew.Sdump()` is mostly used in log statements. When printing command output we usually want to have it in a parse-able format such as JSON. Wasn't aware of the context when you asked me about how to print a block :sweat_smile: